From f9816ca6737fca993ee2f9a288d6b5875047121f Mon Sep 17 00:00:00 2001
From: tonishi92 <tatsuo.onishi@gmail.com>
Date: Fri, 2 Feb 2024 16:13:16 +0100
Subject: [PATCH] CORRECTION for OPT_SFC == 2

---
 src/module_sf_noahmplsm.F | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/module_sf_noahmplsm.F b/src/module_sf_noahmplsm.F
index 091cf75..3895889 100644
--- a/src/module_sf_noahmplsm.F
+++ b/src/module_sf_noahmplsm.F
@@ -3894,6 +3894,10 @@ ENDIF   ! CROPTYPE == 0
        ENDIF
      
        IF(OPT_SFC == 2) THEN
+          ! CORRECTION
+          CH = CH * UR
+          CM = CM * UR
+          ! CORRECTION
           CALL SFCDIF2(parameters,ITER   ,Z0M    ,TAH    ,THAIR  ,UR     , & !in
                        ZLVL   ,ILOC   ,JLOC   ,         & !in
                        CM     ,CH     ,MOZ    ,WSTAR  ,         & !in
-- 
2.31.1

